### PR TITLE
Solved some issues, Added some enhancements.

### PR DESCRIPTION
**Pull request by Steven17D**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,71 @@
-__pycache__
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
```
Comment (yoavravid) [inline .gitignore]: Great

File: makefile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+run:
+	python src/main.py bin/chromedriver ${EMAIL} ${PASSWORD}
```
Comment (yoavravid) [inline makefile]: Is this what we wan't?
Should the chromedriver binary be an argument? is it going to change frequently?
Maybe its better to make it configurable.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+# Usage
+Run in the terminal the following line:
+```
+python3 src\main.py <chrome_drive> <username> <password>
+```
+
+# Requirements
+* Python 3
+* Chrome driver - bin\chromedriver.exe
+    + Download at https://sites.google.com/a/chromium.org/chromedriver/home.
```
Comment (yoavravid) [inline README.md]: Explain how to get the driver on both Windows and Linux.
For Linux you can reference the makefile.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,19 @@
-# yad2_bouncer
-A script for bouncing yad2 ads
+# Yad2 Bouncer
+A script for bouncing yad2 ads.
+
+# Usage
```
Comment (yoavravid) [inline README.md]: Add a section or at least mention the a support for a GUI-less Linux environment.

File: makefile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+install: chrome chromedriver
```
Comment (yoavravid) [inline makefile]: Awesome!!! 👍

File: src/main.py [hunk 6]
```diff
@@ -3,17 +3,23 @@
 
 
 def get_arguments():
-	parser = argparse.ArgumentParser()
```
Comment (yoavravid) [inline src/main.py]: See my comment in the makefile regarding the driver binary as an argument.

File: src/main.py [hunk 7]
```diff
@@ -3,17 +3,23 @@
...
-	yad2 = Yad2()
-	yad2.login(credentials.email, credentials.password)
-	yad2.bounce_all_ads()
+    arguments = get_arguments()
+    yad2 = Yad2(arguments.driver_path)
+    try:
+        yad2.login(arguments.email, arguments.password)
+        yad2.bounce_all_ads()
+    except:
+        yad2.get_screenshot_as_file('error.png')
```
Comment (yoavravid) [inline src/main.py]: Very nice. 
Good for both running on a GUI-less Linux and for running it with task-scheduler on Windows!

File: src/main.py [hunk 8]
```diff
@@ -3,17 +3,23 @@
...
 
 def main():
-	credentials = get_arguments()
-	yad2 = Yad2()
-	yad2.login(credentials.email, credentials.password)
-	yad2.bounce_all_ads()
+    arguments = get_arguments()
+    yad2 = Yad2(arguments.driver_path)
+    try:
+        yad2.login(arguments.email, arguments.password)
```
Comment (yoavravid) [inline src/main.py]: How about logging out?
I'd write a context manager for the login function
`with yad2.login(arguments.email, arguments.password):
yad2.bounce_all_ads()`

File: src/yad2.py [hunk 9]
```diff
@@ -1,49 +1,105 @@
...
-				btn.click()
-				print('Bounced Ad!')
-			except Exception:
-				pass
-
-			self._driver.switch_to.default_content()
+    USERNAME_TEXTBOX_ID = 'userName'
+    PASSWORD_TEXTBOX_ID = 'password'
+    SUBMIT_FORM_ID = 'submitLogonForm'
+    ADS_PAGE_LINK_TEXT = u'מכירות'
```
Comment (yoavravid) [inline src/yad2.py]: Is this variable being used?
If so, don't use it. 
If not, remove it.

File: src/yad2.py [hunk 10]
```diff
@@ -1,49 +1,105 @@
...
+    LOGIN_URL = 'https://my.yad2.co.il/login.php'
+
+    def __init__(self, executable_path):
+        options = webdriver.ChromeOptions()
+        if platform == "linux" or platform == "linux2":
```
Comment (yoavravid) [inline src/yad2.py]: Make sure this works on both platforms.

File: src/yad2.py [hunk 11]
```diff
@@ -1,49 +1,105 @@
...
+            options.binary_location = '/usr/bin/google-chrome-stable'
+            options.add_argument('headless')
+        self._driver = webdriver.Chrome(executable_path=executable_path, options=options)
+        logger_handler = logging.FileHandler('yad2.log')
+        logger_formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+        logger_handler.setFormatter(logger_formatter)
+        self._logger = logging.getLogger(__name__)
+        self._logger.setLevel(logging.DEBUG)
+        self._logger.addHandler(logger_handler)
```
Comment (yoavravid) [inline src/yad2.py]: I'd move the logger initiation logic to a separate function

File: src/yad2.py [hunk 12]
```diff
@@ -1,49 +1,105 @@
...
+        self._driver.get(Yad2.LOGIN_URL)
+        username_textbox = self._driver.find_element_by_id(Yad2.USERNAME_TEXTBOX_ID)
+        password_textbox = self._driver.find_element_by_id(Yad2.PASSWORD_TEXTBOX_ID)
+        submit_button = self._driver.find_element_by_id(Yad2.SUBMIT_FORM_ID)
+        username_textbox.send_keys(email)
+        password_textbox.send_keys(password)
+        submit_button.click()
+        self._logger.info('Finished login')
+
+    def iterate_categories(self):
```
Comment (yoavravid) [inline src/yad2.py]: Add some documentation for what this generator does.
I think that it isn't clear that a generator also clicks on buttons and changes the page.

Comment (yoavravid) [inline src/yad2.py]: Great idea and implementation. Makes the code much more generic!

File: makefile (hunk 2 above)
Comment (Steven17D) [reply makefile]: make will run only in linux envirment so the chrome driver will be fixed

**Comment by Steven17D** [comment]
Made some changes according to your note. Added some improvements.

File: src/yad2.py [hunk 13]
```diff
@@ -1,49 +1,148 @@
...
+
+    def __init__(self, executable_path=None):
+        options = webdriver.ChromeOptions()
+        if platform == "linux" or platform == "linux2":
+            options.binary_location = '/usr/bin/google-chrome-stable'
+            options.add_argument('headless')
+
+        chrome_kwargs = {'options': options}
+
+        if executable_path is not None:
```
Comment (yoavravid) [inline src/yad2.py]: why not do 
```suggestion
        if executable_path is not None:
            self._driver = webdriver.Chrome(executable_path)
        else:
            self._driver = webdriver.Chrome()
       
```

Comment (Steven17D) [reply src/yad2.py]: Code duplication.

**Pull request merged**